hivemind/utils/mpfuture.py

```diff
@@ -219,12 +291,15 @@ def exception(self, timeout: Optional[float] = None) -> Optional[BaseException]:
         return self._exception
```

Comment:
Can we replace the part from `elif self._state == base.CANCELLED:` with just `return super().exception()`?